blockchain/state_manager/src/chain_rand.rs

```diff
@@ -233,8 +236,8 @@ where
         round: ChainEpoch,
         entropy: &[u8],
     ) -> anyhow::Result<[u8; 32]> {
-        // XXX: This `block_on` can be removed by adding a runtime Handle to `ChainRand`.
-        task::block_on(self.get_chain_randomness_v2(&self.blks, pers, round, entropy))
+        self.async_handle
+            .block_on(self.get_chain_randomness_v2(&self.blks, pers, round, entropy))
```

Comment:
To be on the safe side, I think we should use `block_in_place` here.

Reply:
Fixed